ASoC: Intel: soc-acpi-intel-lnl-match: add cs42l43 6x cs35l56 support

bardliao opened the pull request:
This add a match for cs42l43 on sdw link 2, 3 cs35l56 on sdw link 1 and
3. All speakers are aggregated.

bardliao committed:
ASoC: Intel: soc-acpi-intel-lnl-match: add cs42l43 6x cs35l56 support

This add a match for cs42l43 on sdw link 2, 3 cs35l56 on sdw link 1 and
3. All speakers are aggregated.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

bardliao commented:
@simontrimmer @rfvirgil I vaguely remember that you have a naming rule for acpi adr devices but I can't find it. Could you correct me if `cs35l56_1_3amp_adr[]` against the naming rule?

simontrimmer commented:
Hi @bardliao I think the naming convention recollection you may be thinking of was this one:

> commit a3003af649efb6f3d86d379d1e9a966ea6d5f5ab
> Author: Simon Trimmer <simont@opensource.cirrus.com>
> Date:   Fri Dec 6 15:59:02 2024 +0800
> 
>     ASoC: Intel: soc-acpi: arl: Correct naming of a cs35l56 address struct
> 
>     As there are many combinations these follow a naming scheme to make
>     the content of link structures clearer:
> 
>     cs35l56_<controller link>_<l or r><unique instance id>_adr
> 
>     Signed-off-by: Simon Trimmer <simont@opensource.cirrus.com>
>     Reviewed-by: Péter Ujfalusi <peter.ujfalusi@linux.intel.com>
>     Reviewed-by: Liam Girdwood <liam.r.girdwood@intel.com>
>     Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>
>     Link: https://patch.msgid.link/20241206075903.195730-10-yung-chuan.liao@linux.intel.com
>     Signed-off-by: Mark Brown <broonie@kernel.org>

Will have a look at the naming in a mo
-Simon

simontrimmer commented on sound/soc/intel/common/soc-acpi-intel-lnl-match.c:
```diff
@@ -159,6 +187,33 @@ static const struct snd_soc_acpi_endpoint cs42l43_endpoints[] = {
 	},
 };
 
+static const struct snd_soc_acpi_endpoint cs42l43_amp_spkagg_ndpoints[] = {
```
cs42l43_amp_spkagg_ndpoints -> cs42l43_amp_spkagg_endpoints ?

simontrimmer commented on sound/soc/intel/common/soc-acpi-intel-lnl-match.c:
```diff
@@ -204,6 +259,48 @@ static const struct snd_soc_acpi_adr_device cs35l56_3_lr_adr[] = {
+	}
+};
+
+static const struct snd_soc_acpi_adr_device cs35l56_3_3amp_adr[] = {
```
Thinking about that card I don't think we have a concept of a left and a right so it's just the "3amp" as the unique identifier which is fine by me

lgirdwood replied:
Should we call this assumption out as an inline comment.

bardliao replied:
@lgirdwood There is no assumption here. We introduce the amp's adr and their positions here. The naming here is to make sure we will have unique name for each acpi adr device array.

bardliao committed:
ASoC: Intel: soc-acpi-intel-ptl-match: add cs42l43 6x cs35l56 support

This add a match for cs42l43 on sdw link 2, 3 cs35l56 on sdw link 1 and
3. All speakers are aggregated.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

bardliao commented:
@simontrimmer Does it look good now?

simontrimmer commented:
Hi @bardliao I don't have one of those boards but yes looks good to me!
